From 9d294341f62be3d3e2716eb6111cfe61a21d9c99 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Fri, 26 Apr 2013 16:54:22 +0200
Subject: [LIBREPORT PATCH 1/6] reporter-bugzilla: support new format keyword
 %reporter

Readd lost information essential for bug fixing.

Example of line added to a bug report:
reporter:    libreport-2.1.3.36.g5816.dirty

Related to #155

Signed-off-by: Jakub Filak <jfilak@redhat.com>
Signed-off-by: Martin Milata <mmilata@redhat.com>
---
 src/plugins/bugzilla_format.conf           | 1 +
 src/plugins/bugzilla_format_kernel.conf    | 2 +-
 src/plugins/bugzilla_format_libreport.conf | 1 +
 src/plugins/bugzilla_formatdup.conf        | 1 +
 src/plugins/reporter-bugzilla.c            | 4 ++++
 5 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/plugins/bugzilla_format.conf b/src/plugins/bugzilla_format.conf
index b06e68a..f668d17 100644
--- a/src/plugins/bugzilla_format.conf
+++ b/src/plugins/bugzilla_format.conf
@@ -48,6 +48,7 @@ Additional info:: \
 	-analyzer,-count,-duphash,-uuid,-abrt_version,\
 	-username,-hostname,-os_release,-os_info,\
 	-time,-pid,-pwd,\
+	%reporter,\
 	%oneline
 
 Truncated backtrace:: %bare_%short_backtrace
diff --git a/src/plugins/bugzilla_format_kernel.conf b/src/plugins/bugzilla_format_kernel.conf
index c32e424..e46af89 100644
--- a/src/plugins/bugzilla_format_kernel.conf
+++ b/src/plugins/bugzilla_format_kernel.conf
@@ -40,6 +40,6 @@
 
 Description of problem:: %bare_comment
 
-Additional info:: %bare_backtrace
+Additional info:: %reporter, %bare_backtrace
 
 %attach:: dmesg
diff --git a/src/plugins/bugzilla_format_libreport.conf b/src/plugins/bugzilla_format_libreport.conf
index e5b3870..40428a5 100644
--- a/src/plugins/bugzilla_format_libreport.conf
+++ b/src/plugins/bugzilla_format_libreport.conf
@@ -50,6 +50,7 @@ Additional info:: \
 	-analyzer,-count,-duphash,-uuid,-abrt_version,\
 	-username,-hostname,-os_release,-os_info,\
 	-time,-pid,-pwd,\
+	%reporter,\
 	%oneline
 
 Truncated backtrace:: %bare_%short_backtrace
diff --git a/src/plugins/bugzilla_formatdup.conf b/src/plugins/bugzilla_formatdup.conf
index 2ec183e..99f63c2 100644
--- a/src/plugins/bugzilla_formatdup.conf
+++ b/src/plugins/bugzilla_formatdup.conf
@@ -49,4 +49,5 @@
 	-username,-hostname,-os_release,-os_info,\
 	-time,-pid,-pwd,\
 	-var_log_messages,\
+	%reporter,\
 	%oneline
diff --git a/src/plugins/reporter-bugzilla.c b/src/plugins/reporter-bugzilla.c
index e9e0a70..d0e04b6 100644
--- a/src/plugins/reporter-bugzilla.c
+++ b/src/plugins/reporter-bugzilla.c
@@ -436,6 +436,10 @@ int append_item(struct strbuf *result, const char *item_name, problem_data_t *pd
     if (strcmp(item_name, "%short_backtrace") == 0)
         return append_short_backtrace(result, pd, CD_TEXT_ATT_SIZE_BZ, print_item_name);
 
+    /* Compat with previously-existed ad-hockery: %reporter */
+    if (strcmp(item_name, "%reporter") == 0)
+        return append_text(result, "reporter", PACKAGE"-"VERSION, print_item_name);
+
     /* %oneline,%multiline,%text */
     bool oneline   = (strcmp(item_name+1, "oneline"  ) == 0);
     bool multiline = (strcmp(item_name+1, "multiline") == 0);
-- 
1.8.1.4

